### PR TITLE
Use Python3 super()

### PR DESCRIPTION
**Pull request by jdufresne**
Follows commit fbe9620784b130dcdccf43cd3105940152c4d762.

### CODE REVIEW

File: src/urllib3/connection.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import re
 import socket
 import warnings
-from http.client import HTTPConnection as _HTTPConnection
```
Comment (sethmlarson) [inline src/urllib3/connection.py:7]: Can we revert the change here? Ensures that HTTPConnection name can only be one thing throughout this file.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Instead of doing this `super()` upgrade file by file can you do it all in one PR if there's upgrades to be done elsewhere too?

File: src/urllib3/connection.py (hunk 1 above)
Comment (jdufresne) [reply src/urllib3/connection.py]: done.

**Comment by jdufresne** [comment]
I think I got them all, but please let me know if you notice any more.

**Review by sethmlarson — APPROVED** [review]
LGTM, thanks!
